Plans on site domain display updated card

ravishanker opened the pull request:
Fixes #19807, #19808, #19809   

Updates card with `upgrade to a plan` CTA, besides existing `get a domain` CTA on domains dashboard for those on free plan .

| Before | After |
|--------|--------|
| ![Screenshot_20231219_193130](https://github.com/wordpress-mobile/WordPress-Android/assets/990349/1a127106-0ed5-495e-ba72-22b9e91294e8) |![Screenshot_20231220_112652](https://github.com/wordpress-mobile/WordPress-Android/assets/990349/63988d4a-defd-4d28-b6ca-953d2470b362) | 

-----

## To Test:

Test 1
- Launch JP app
- `Select` a site with a free plan, have admin role
- `Go` to **Domains** (My site -> domains or My site -> More... -> domain)
- Verify a new card is displayed as in second image above
- `Tap` on **Upgrade to a plan**
- `Verify` it launches plans purchase flow like plans card on dashboard
- Return to domains
- `Tap` on **Just search for a domain**
- `Verify` it launches domain purchase flow as before

Test 2
- For completeness, also verify with a site that has a paid plan
- `Verify` it doesn't display the card above, but the card as before

Test 3
- `Verify` analytics tracker in AS like 
🔵 Tracked: domains_dashboard_get_plan_tapped, Properties: {"blog_id":196875124,"site_type":"blog","is_jetpack":false}
-----

## Regression Notes

1. Potential unintended areas of impact

    - None

2. What I did to test those areas of impact (or what existing automated tests I relied on)

    - Manual, and unit tests

3. What automated tests I added (or what prevented me from doing so)

    - Updated unit tests

-----

## PR Submission Checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

ravishanker committed:
Create domain_plan_purchase_card.xml

ravishanker committed:
Update dimens.xml

ravishanker committed:
Update strings.xml

ravishanker committed:
Update DomainsDashboardItem.kt

ravishanker committed:
Update DomainsDashboardViewHolder.kt

ravishanker committed:
Update DomainsDashboardNavigationAction.kt

ravishanker committed:
Update DomainsDashboardAdapter.kt

ravishanker committed:
Update DomainsDashboardViewModel.kt

ravishanker committed:
Update DomainsDashboardFragment.kt

ravishanker committed:
Update DomainsDashboardViewModelTest.kt

irfano reviewed (changes requested):
The code change LGTM! 👍🏻 
Also, we may need to change the design depending on the discussion: p1702984046790589-slack-C06AMD6LSPN

The only issue I discovered is on the horizontal mode, sharing here:
<img src="https://github.com/wordpress-mobile/WordPress-Android/assets/2471769/b38ce8e6-db26-47e5-96ba-3baa858033ba">

ravishanker committed:
Update card to match latest design

- Updated domains plan card layout, styles and text

ravishanker committed:
Update for landscape orientation

Fix layout, and styles for landscape orientation

ravishanker commented:
> The code change LGTM! 👍🏻 Also, we may need to change the design depending on the discussion: p1702984046790589-slack-C06AMD6LSPN
> 
> The only issue I discovered is on the horizontal mode

Thanks Irfan.  Update to the latest design, fixed landscape orientation, added tracking, and updated release notes as well to save time.

ravishanker committed:
Remove unused dimen

ravishanker committed:
Add tracking to Upgrade to a plan button

ravishanker committed:
Update RELEASE-NOTES.txt

Update for Upgrade to a plan from domains dashboard in v24.0

irfano committed:
Update sizes in domain_plan_purchase_card according to the new design

irfano committed:
Update text style in domain_plan_purchase_card according to new design

irfano commented:
The layout issue has been resolved with your commit. 👍🏻 
The new design looks good, even though you didn't have access to the Figma file, your design is pretty close it. I made minor updates to the style to align with the Figma file.

ravishanker committed:
Merge branch 'trunk' into Plans-on-site-domain-display-updated-card
